CMakeLists.txt

```diff
@@ -282,22 +295,39 @@ if(WITH_SPDK)
   set(HAVE_SPDK TRUE)
 endif(WITH_SPDK)
 
-option(WITH_PMEM "Enable PMEM" OFF)
-if(WITH_PMEM)
-  set(HAVE_PMEM ON)
-  if(NOT WITH_BLUESTORE)
-    message(SEND_ERROR "Please enable WITH_BLUESTORE for using PMEM")
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i686|amd64|x86_64")
+  # NVML is now PMDK
+  if(WITH_PMEM_DEVICE)
+    option(WITH_PMDK "Enable PMDK libraries" ON)
+  else()
+    option(WITH_PMDK "Enable PMDK libraries" OFF)
+  endif(WITH_PMEM_DEVICE)
```

Comment:
iiuc, if `WITH_PMDK` is an alias of `WITH_PMEM_DEVICE`, why shall we have both of them? or, is there any development kit/library else than PMDK that be used to program PMEM devices?

Reply:
IIRC, I think ```WITH_PMDK``` will eventually support both ```WITH_PMEM_DEVICE``` (OSD) and ```WITH_RBD_RWL``` (RBD writeback cache stored using PMDK)

Reply:
@dillaman is right. This is the first PR of RBD writeback cache which also uses PMDK. A later patch will add another option.

Reply:
could you replace `endif(WITH_PMEM_DEVICE)` with `endif()`?

Reply:
This part is replaced by CMAKE_DEPENDENT_OPTION